src/worker.rs

```diff
@@ -173,25 +173,28 @@ impl Worker {
         allowed_to_mutate: bool,
         validation_response: ValidationResponse,
     ) -> ValidationResponse {
-        if validation_response.patch.is_some() && !allowed_to_mutate {
-            // Return early -- a policy not allowed to mutate tried to
-            // mutate --, we don't care about the policy mode at this
-            // point, this will be a problem in Protect mode, so
-            // return the error too if we are in Monitor mode.
-            return ValidationResponse {
-                allowed: false,
-                status: Some(ValidationResponseStatus {
-                    message: Some(format!("Request rejected by policy {}. The policy attempted to mutate the request, but it is currently configured to not allow mutations.", policy_id)),
-                    code: None,
-                }),
-                // if `allowed_to_mutate` is false, we are in a validating webhook. If we send a patch, k8s will fail because validating webhook do not expect this field
-                patch: None,
-                patch_type: None,
-                ..validation_response
-            };
-        };
         match policy_mode {
-            PolicyMode::Protect => validation_response,
+            PolicyMode::Protect => {
+                if validation_response.patch.is_some() && !allowed_to_mutate {
+                    // Return early -- a policy not allowed to mutate tried to
+                    // mutate --, we don't care about the policy mode at this
+                    // point, this will be a problem in Protect mode, so
+                    // return the error too if we are in Monitor mode.
```

Comment:
nit update these lines because they mention the protect/monitor mode. Now this block of code is always in "protect" mode. I think you can even drop these lines to be honest...

Reply:
Done